onnxruntime/core/optimizer/embed_layer_norm_fusion.cc

```diff
@@ -65,65 +65,149 @@ static bool CheckInput(NodeArg* input, const logging::Logger& logger) {
+
+  if (position_gather_node.OpType() != "Gather") {
     return false;
   }
-  if (edges[0]->GetNode().GetOutputEdgesCount() != 1 && edges[1]->GetNode().GetOutputEdgesCount() != 1) {
+  std::vector<const Node::EdgeEnd*> pg_edges;
+  // Find the "Expand" node
+  if (!graph_utils::FindPath(position_gather_node, true, {{0, 1, "Expand", {8}, kOnnxDomain}}, pg_edges, logger)) {
     return false;
   }
-
-  // Match Shape --> Gather --> Unsqueeze --> ConstantOfShape --> NonZero --> Transpose --> Squeeze --> Cast --> Unsqueeze --> Expand
-  Node& expand_node = *graph.GetNode(edges[0]->GetNode().Index());
-  Node& shape_node_1 = *graph.GetNode(edges[1]->GetNode().Index());
-  std::vector<graph_utils::EdgeEndToMatch> pg_parent_path{
-      {0, 0, "Unsqueeze", {1, 11}, kOnnxDomain},
-      {0, 0, "Cast", {9}, kOnnxDomain},
-      {0, 0, "Squeeze", {1}, kOnnxDomain},
-      {0, 0, "Transpose", {1}, kOnnxDomain},
-      {0, 0, "NonZero", {9}, kOnnxDomain},
-      {0, 0, "ConstantOfShape", {9}, kOnnxDomain},
-      {0, 0, "Unsqueeze", {1, 11}, kOnnxDomain},
-      {0, 0, "Gather", {1, 11}, kOnnxDomain},
-      {0, 0, "Shape", {1}, kOnnxDomain},
-  };
-  matched_edges = edges;
-
-  if (!graph_utils::FindPath(expand_node, true, pg_parent_path, edges, logger)) {
+  Node& expand_node = *graph.GetNode(pg_edges[0]->GetNode().Index());
+  const Node::EdgeEnd* expand_edge = pg_edges[0];
+  // Look for Path 1:
+  // Shape --> Gather --> Unsqueeze --> ConstantOfShape --> NonZero --> Transpose --> Squeeze --> Cast --> Unsqueeze --> Expand
+  if (!graph_utils::FindPath(expand_node, true,
+                             {{0, 0, "Unsqueeze", {1, 11}, kOnnxDomain},
+                              {0, 0, "Cast", {9}, kOnnxDomain},
+                              {0, 0, "Squeeze", {1}, kOnnxDomain},
+                              {0, 0, "Transpose", {1}, kOnnxDomain},
+                              {0, 0, "NonZero", {9}, kOnnxDomain},
+                              {0, 0, "ConstantOfShape", {9}, kOnnxDomain},
+                              {0, 0, "Unsqueeze", {1, 11}, kOnnxDomain},
+                              {0, 0, "Gather", {1, 11}, kOnnxDomain},
+                              {0, 0, "Shape", {1}, kOnnxDomain}},
+                             pg_edges, logger)) {
     return false;
   }
-  for (size_t i = 0; i < edges.size(); i++) {
-    if (edges[i]->GetNode().GetOutputEdgesCount() != 1) {
+  // All nodes in Path 1 except the "Gather" node must have only 1 output edge.
+  for (size_t i = 0; i < pg_edges.size() - 2; i++) {
+    if (pg_edges[i]->GetNode().GetOutputEdgesCount() != 1) {
       return false;
     }
   }
+  if (pg_edges[8]->GetNode().GetOutputEdgesCount() != 1) {
+    return false;
+  }
   // Check if the second input of the Gather node in the path has a constant input of 1
-  Node& gather_node = *graph.GetNode(edges[edges.size() - 2]->GetNode().Index());
+  Node& gather_node = *graph.GetNode(pg_edges[pg_edges.size() - 2]->GetNode().Index());
```

Comment:
how to logic like:

if (gather node has two output edges) {
  match new sub-graph.
}
else if (has one output edges) {
  match old sub-graph.
}

It is because the new sub-graph is same as the one in opset 11. We can refactor it to remove common code later.